client/client.go

```diff
@@ -217,6 +220,14 @@ func Insecurely() Option {
 	}
 }
 
+// DecouplePrevSig
+func DecouplePrevSig() Option {
```

Comment:
I'm thinking we should treat the whole thing not as just a configuration, but a new version of a protocol, to make it extremely clear this is an incompatible protocol as before, and not just a tweak. 
As such I would call this V2() for example with a clear comment as to what V2 means.

This is an exercise as well to see how having multiple protocol can work, how we manage different versions.